gilbreth_perception/src/recognition_node.cpp

```diff
@@ -67,10 +70,12 @@ void loadParameter() {
   max_correspondence_distance = parameter_map["max_correspondence_distance"];
   nr_iterations = parameter_map["nr_iterations"];
   visualizer = switch_map["visualizer"];
+  icp = switch_map["ICP"];
 }
 
-void loadModel(std::vector<std::vector<double> > &pick_pose, std::vector<std::string> &model_name,
-               std::vector<pcl::PointCloud<PointType>::Ptr> &model_list, std::vector<pcl::PointCloud<pcl::FPFHSignature33>::Ptr> &model_features_list) {
+void loadModel(std::vector<std::vector<double>> &pick_pose, std::vector<std::string> &model_name,
+               std::vector<pcl::PointCloud<PointType>::Ptr> &model_list, std::vector<pcl::PointCloud<pcl::FPFHSignature33>::Ptr> &model_features_list,
+               std::vector<pcl::PointCloud<pcl::SHOT352>::Ptr> &model_descriptor_list, std::vector<pcl::PointCloud<pcl::ReferenceFrame>::Ptr> &model_rf_list) {
```

Comment:
To make the code cleaner, you create a ```loadICPConfig``` and ```loadCGConfig``` and call one or the other depending on what's chosen.